generator/cmd/generator/main.go

```diff
@@ -21,20 +45,34 @@ func (c Creator) Create(name string) (io.WriteCloser, error) {
 	return os.Create(abs)
 }
 
-//go:generate go run main.go --clean --schema ../../schema.json --output-directory ../../../graph_objects
-
-func main() {
-	clean := flag.Bool("clean", false, "clean the output directory first. Mandatory on CI")
-	schema := flag.String("schema", "schema.json", "plotly schema")
-	outputDirectory := flag.String("output-directory", "gen/", "output directory, must exist before generation")
+// FindDir goes back in the project directory to find the correct location of the schema and output folders
+// this allows the script to run smoothlessly with go generate in the cmd files,
+// and also when it is run from the project root as working directory
+func FindDir(schema, output string) (string, string) {
+	for i := 0; i < 4; i++ {
+		if _, err := os.Stat(schema); err == nil {
+			break
+		}
+		schema = filepath.Join("../", schema)
+		output = filepath.Join("../", output)
+	}
+	return schema, output
+}
```

Comment:
This is risky, if there is a bug here and you combine it with the RemoveAll.... you may wipe your entire disk.

So I refactor this to take the root of the schema as the base. 

Also I added the clean flag again for the same reason, I don't want a bug to accidentally wipe the entire disk, it is something I want people to execute carefully. If possible, only on CI

Reply:
thanks for your improvements